src/common/pick_address.cc

```diff
@@ -648,10 +648,16 @@ bool is_addr_in_subnet(
   ceph_assert(!nets.empty());
 
   unsigned ipv = CEPH_PICK_ADDRESS_IPV4;
-  struct sockaddr_in public_addr;
-  public_addr.sin_family = AF_INET;
-
-  if(inet_pton(AF_INET, addr.c_str(), &public_addr.sin_addr) != 1) {
+  struct sockaddr_in6 public_addr6;
+  struct sockaddr_in public_addr4;
+
+  if (inet_pton(AF_INET, addr.c_str(), &public_addr4.sin_addr) == 1) {
+    ipv = CEPH_PICK_ADDRESS_IPV4;
+    public_addr4.sin_family = AF_INET;
+  } else if (inet_pton(AF_INET6, addr.c_str(), &public_addr6.sin6_addr) == 1) {
+    ipv = CEPH_PICK_ADDRESS_IPV6;
+    public_addr6.sin6_family = AF_INET6;
+  } else {
```

Comment:
`is_addr_in_subnet` is called from `OSDMap::get_out_of_subnet_osd_counts`.
Instead of passing the address as a string (using `ip_only_to_str`) and parsing it again here, I think that we can pass the address as `entity_addr_t` which already supports `is_ipv4()` and `is_ipv6()`.

Reply:
nit: commit message states - Improved logging to handle invalid or unsupported addresses more clearly.